test/ok/error_test.exs

```diff
@@ -0,0 +1,35 @@
+defmodule OK.ErrorTest do
+  use ExUnit.Case
+
+  def foo(x) do
+    {:ok, x}
+  end
+
+  def bar(y) do
+    {:bad, y}
+  end
+
+  require OK
+
+  test "helpful error for binding failure" do
+    message = """
+    Binding to variable failed, '{:bad, 6}' is not a result tuple.
+
+        Code
+          b <- bar(a)
+
+        Expected signature
+          bar(a) :: {:ok, b} | {:error, reason}
+
+        Actual values
+          bar(a) :: {:bad, 6}
+    """
+    assert_raise OK.BindError, message, fn() ->
+      OK.with do
+        a <- foo(6)
+        b <- bar(a)
+        _ <- {:ok, b}
```

Comment:
> This compiler error was always there. it is the same situation as you fixed earlier but in a slightly different place because you fixed it when the last line was not a bind.

I'm bringing it to your attention, as compiler warnings is one of things I look for when I'm checking out a third-party lib. Previously, you addressed this particular error with an anonymous wrapper. I tried that this time and it still gave the compiler warning.

> Most people would probably write
> ...
> But maybe it is worth fixing.

I would say that the unit test should be what the recommended best practice is, which IMO, is returning `OK.success` explicitly. This gives the same visual cue that the native `with`'s `do` section gives, allowing for self-documenting code. At the very least it could be changed to the raw `:ok` tuple as you mention. Either way, doing this will both show your recommended best practice _and_ get rid of the compiler warning.

> 2. This case clause error is from the case clause at line 221. It's technically not a bind error, there is no bind in that line. It is a return value error because with requires you to return a result as the last line. I was tempted to lift this restriction.

I would think the error should be the same throughout, including this last statement, since the construct is enforcing its ultimate return value be a result monad. With this in mind, perhaps this means that the name `BindError` is actually a `ResultMonadError`? Or maybe the last line is an implicit bind. 

The reasoning is definitely more in your field, I'm just saying that to me this seems like an inconsistency and I would be confused if I saw the error - especially if I were aware of the `BindError`'s existence.